Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,263 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ForgeSchemaValidator.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The ForgeSchemaValidator class.
+// </summary>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Forge.TreeWalker
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using Microsoft.Forge.DataContracts;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using Newtonsoft.Json.Schema;
+
+    /// <summary>
+    /// The ForgeSchemaValidator class implements the validation method that tests input schemas with input custom rules.
+    /// </summary>
+    public static class ForgeSchemaValidator
+    {
+
+        /// <summary>
+        /// Linked rules in JSchema type.
```

Comment:
>Linked rules in JSchema type. [](http://example.com/codeflow?start=12&length=29)

"
Links parent JSON schema validation rules with child rules through the referenceUri.
See ForgeSchemaValidatorTests for examples.
"